src/refmt_impl.ml

```diff
@@ -138,118 +114,48 @@ let () =
-        | Some "re" -> Reason_toolchain.JS.print_canonical_interface_with_comments
-        | Some s -> (
-          raise (Invalid_config ("Invalid --print setting for interface '" ^ s ^ "'."))
-        )
-      in
-      thePrinter (ast, comments)
-    ) else (
-      let ((ast, comments), parsedAsML, parsedAsInterface) = match !prse with
-        | None -> (defaultImplementationParserFor use_stdin filename)
-        | Some "binary_reason" -> reasonBinaryParser use_stdin filename
-        | Some "binary" -> ocamlBinaryParser use_stdin filename false
-        | Some "ml" -> (Reason_toolchain.ML.canonical_implementation_with_comments (Reason_toolchain.setup_lexbuf use_stdin filename), true, false)
-        | Some "re" -> (Reason_toolchain.JS.canonical_implementation_with_comments (Reason_toolchain.setup_lexbuf use_stdin filename), false, false)
-        | Some s -> (
-          raise (Invalid_config ("Invalid --parse setting for interface '" ^ s ^ "'."))
-        )
-      in
-      let _ =
-          if parsedAsInterface then
-              raise (Invalid_config ("The file parsed does not appear to be an implementation file."))
-      in
-      let _ = Reason_pprint_ast.configure
-          ~width: print_width
-          ~assumeExplicitArity: !assumeExplicitArity
-          ~constructorLists
-      in
-      let thePrinter = match !prnt with
-        | Some "binary_reason" -> fun (ast, comments) -> (
-          (* Our special format for interchange between reason should keep the
-           * comments separate.  This is not compatible for input into the
-           * ocaml compiler - only for input into another version of Reason. We
-           * also store whether or not the binary was originally *parsed* as an
-           * interface file.
-           *)
-          output_value stdout (
-            Config.ast_impl_magic_number, filename, ast, comments, parsedAsML, false
-          );
-        )
-        | Some "binary"
-        | None -> fun (ast, comments) -> (
-          output_string stdout Config.ast_impl_magic_number;
-          output_value  stdout filename;
-          output_value  stdout ast
-        )
-        | Some "ast" -> fun (ast, comments) -> (
-          Printast.implementation Format.std_formatter ast
-        )
-        (* If you don't wrap the function in parens, it's a totally different
-         * meaning #thanksOCaml *)
-        | Some "none" -> (fun (ast, comments) -> ())
-        | Some "ml" -> Reason_toolchain.ML.print_canonical_implementation_with_comments
-        | Some "re" -> Reason_toolchain.JS.print_canonical_implementation_with_comments
-        | Some s -> (
-          raise (Invalid_config ("Invalid --print setting for implementation '" ^ s ^ "'."))
-        )
-      in
-      thePrinter (ast, comments)
+    let (ast, parsedAsML) = Printer.parse !prse use_stdin filename in
+    (* As it turns out, calling formatter_of_out_channel on stdout results in
```

Comment:
Ah, whoops. Did not mean to leave this comment in. I think I mislabeled the true culprit.